experiments/testing_archive/test_experiment_exceptions.py

```diff
@@ -0,0 +1,151 @@
+"""
+Test module for the experiment_handler/experiment_prototype code.
+It is run simply via 'python3 test_experiment_exceptions.py' and will go through all tests
+in the experiment_tests.csv file as well as the hardcoded tests here that don't fit nicely into
+a csv file.
+
+The csv file format is:
+[#][experiment file module import name]::[regex error message]
+
+The [#] is an optional comment, and that line will be removed
+An example of a test line is:
+testing_archive.my_test_experiment.py::Regex line that * matches the ExperimentException err msg
+
+References:
+https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python
+https://docs.python.org/3/library/unittest.html
+
+:copyright: 2020 SuperDARN Canada
+:author: Kevin Krieger
+"""
+
+
+import unittest
+import os
+import sys
+import argparse
+
+BOREALISPATH = os.environ['BOREALISPATH']
+sys.path.append(BOREALISPATH)
+# Need to hardcode this, as unittest does weird things when you supply an argument on command line,
+# or if you use argparse. There is probably a better way
+input_test_file = "PATH_TO_/experiment_tests.csv"
+
+# Call experiment handler main function like so: eh.main(['normalscan', 'discretionary'])
+import experiment_handler.experiment_handler as eh
+from experiment_prototype.experiment_exception import ExperimentException
+
+
+def ehmain(experiment='normalscan', scheduling_mode='discretionary'):
+    """
+    Convenience method to call the experiment handler with arguments
+    """
+    eh.main([experiment, scheduling_mode])
+
+
+class TestExperimentEnvSetup(unittest.TestCase):
+    """
+    A unittest class to test the environment setup for the experiment_handler module.
+    All test methods must begin with the word 'test' to be run by unittest.
+    """
+    def setUp(self):
+        """
+        This function is called before every test_* method (every test case in unittest lingo)
+        """
+        print("Method: ", self._testMethodName)
+
+    def test_no_args(self):
+        """
+        Test calling the experiment handler without any command line arguments, which returns 2
+        """
+        with self.assertRaisesRegex(SystemExit, "2"):
+            eh.main([])
+
+    def test_borealispath(self):
+        """
+        Test failure to have BOREALISPATH in env
+        """
+        # Need to remove the environment variable, reset for other tests
+        del os.environ['BOREALISPATH']
+        with self.assertRaisesRegex(KeyError, "BOREALISPATH"):
+            ehmain()
+        os.environ['BOREALISPATH'] = BOREALISPATH
+
+    def test_config_file(self):
+        """
+        Test the code that checks for the config file
+        """
+        # Rename the config file temporarily
+        os.rename(BOREALISPATH + 'config.ini', BOREALISPATH + '_config.ini')
```

Comment:
I had to do this to get the tests to work on my system ; should also work if your path ends in a /
```suggestion
        os.rename(BOREALISPATH + '/config.ini', BOREALISPATH + '/_config.ini')
```